src/component.jl

```diff
@@ -184,29 +195,120 @@ Base.sortperm(c::SharedComponent) = sortperm(c.data)
     end
     return h
 end
+
+#TODO: These boundchecks etc are to ensure in the case of || in @entities_in nobody is trying to load some
+#      invalid memory. Do we really need ||?
```

Comment:
Personally I hardly use `||`.

But it should also be possible to emit a different type of iterator when `||` is used, and only have the bounds checks on in those cases.

Reply:
Having tested it a bit more on an actual "real" world-y example (see test/benchmark.jl), the difference of having that boundscheck vs not having it is 48.23 microsec vs 47.78microsec, maybe that's acceptable for now (although it itches a bit and I will put a todo :sweat_smile:).

Reply:
Sounds fine for the moment then!